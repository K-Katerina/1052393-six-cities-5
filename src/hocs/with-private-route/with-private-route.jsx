import React from "react";
import {Redirect} from "react-router-dom";
import PropTypes from "prop-types";

export const withPrivateRoute = (Component, isLoggedIn, URL = `/`) => (props) => {
  if (isLoggedIn) {
    return <Component {...props}/>;
  } else {
    return <Redirect to={URL}/>;
  }
};

withPrivateRoute.propTypes = {
  loggedIn: PropTypes.bool.isRequired
};
