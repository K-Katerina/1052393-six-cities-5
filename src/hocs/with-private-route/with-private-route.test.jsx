import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import {Provider} from "react-redux";
import {createStore} from "redux";
import rootReducer from "../../store/reducers/root-reducer";
import {BrowserRouter} from "react-router-dom";
import {withPrivateRoute} from "./with-private-route";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

const MockComponentWrapped = withPrivateRoute(MockComponent);

it(`withPrivateRoute is render correctly`, () => {
  const tree = renderer.create((
    <Provider store={createStore(rootReducer)}>
      <BrowserRouter>
        <MockComponentWrapped
          loggedIn={true}
        >
          <React.Fragment />
        </MockComponentWrapped>
      </BrowserRouter>
    </Provider>
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
