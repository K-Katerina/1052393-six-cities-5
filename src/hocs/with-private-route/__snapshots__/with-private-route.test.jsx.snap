// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withPrivateRoute is render correctly 1`] = `null`;
